### PR TITLE
feat: Create skeleton component

### PR DESCRIPTION
**Pull request by FDELAHA24**
## Changes description 🧑‍💻

Add skeleton components

## Context 🤔

[#3](https://github.com/Decathlon/vitamin-compose/issues/3)

## Checklist ✅

- [x] I have reviewed the submitted code
- [x] I have tested on a phone device/emulator
- [x] I have tested on a tablet device/emulator
- [x] I have tested on a kiosk device/emulator

#### Phone

Screenshot_20220617_143605

#### Tablet

Screenshot_20220617_143834

#### Kiosk

Screenshot_20220617_144019

### CODE REVIEW

**Comment by dktbitrise** [comment]
### You can download the apk here :
**sample-debug.apk :** 
https://app.bitrise.io/app/e9ae480151be4514/build/6916b172-d8f5-4e28-82c1-bf663fc0537b/artifact/6b9f906ea2e36f4b/p/83095b5d8aa3d7d34acdf4939d2b219f

**Comment by dktbitrise** [comment]
### You can download the apk here :
**sample-debug.apk :** 
https://app.bitrise.io/app/e9ae480151be4514/build/45ce3f7d-c1ae-4dfb-b418-0576cf6e21fb/artifact/17a8707b24a67ef9/p/e3dc7e462b86ecf02b17e9c37e83cfa3

**Comment by N0t-Simon** [comment]
✅

**Comment by N0t-Simon** [comment]
✅

**Comment by dktbitrise** [comment]
### You can download the apk here :

**Comment by dktbitrise** [comment]
### You can download the apk here :
**sample-debug.apk :** 
https://app.bitrise.io/app/e9ae480151be4514/build/b919ee01-2742-47cb-80bc-af1c3b3335a7/artifact/1d36710d79265dc1/p/b3e3736f5ac9b76409738b76bd3a9a42

File: skeleton/src/main/java/com/decathlon/vitamin/compose/skeleton/VerticalShimmer.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import androidx.annotation.FloatRange
+import androidx.compose.animation.core.InfiniteRepeatableSpec
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.util.lerp
+import com.google.accompanist.placeholder.PlaceholderDefaults
+import com.google.accompanist.placeholder.PlaceholderHighlight
+
+fun PlaceholderHighlight.Companion.verticalShimmer(
```
Comment (GerardPaligot) [inline skeleton/src/main/java/com/decathlon/vitamin/compose/skeleton/VerticalShimmer.kt:12]: It should be `internal` no?

File: skeleton/src/main/java/com/decathlon/vitamin/compose/skeleton/VitaminSkeleton.kt [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+     * @param color color of this skeleton shape
+     * @param highlightColor color of the shimmer effect
+     * @param cornerRadius size of the rectangle shape corner radius, no radius by default
+     * @param contentDescription the [String] for the content description
+     */
+    @Composable
+    fun Rectangular(
+        modifier: Modifier = Modifier,
+        color: Color = VitaminTheme.colors.vtmnBackgroundTertiary,
+        highlightColor: Color = VitaminTheme.colors.vtmnBackgroundPrimary,
```
Comment (GerardPaligot) [inline skeleton/src/main/java/com/decathlon/vitamin/compose/skeleton/VitaminSkeleton.kt:33]: Same here (and for all components in this object), we should probably used a class for all colors.

File: skeleton/src/main/java/com/decathlon/vitamin/compose/skeleton/VitaminSkeleton.kt [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+private fun VitaminSkeletonShapeImpl(
+    color: Color,
+    highlightColor: Color,
+    shape: Shape,
+    modifier: Modifier,
+    contentDescription: String = stringResource(R.string.vtmn_cd_is_loading)
+) {
+    Box(
+        modifier = modifier
+            .placeholder(
```
Comment (GerardPaligot) [inline skeleton/src/main/java/com/decathlon/vitamin/compose/skeleton/VitaminSkeleton.kt:76]: You don't use your own modifier `vtmnSkeleton`? :o

**Review by GerardPaligot — CHANGES_REQUESTED** [review]

File: skeleton/src/main/java/com/decathlon/vitamin/compose/skeleton/SkeletonMofifierExt.kt [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+ * @param highlightColor color of the shimmer effect
+ * @param contentDescription the [String] for the content description
+ */
+@Composable
+fun Modifier.vtmnSkeleton(
+    visible: Boolean = true,
+    color: Color = VitaminTheme.colors.vtmnBackgroundTertiary,
+    shape: Shape = RectangleShape,
+    highlightColor: Color = VitaminTheme.colors.vtmnBackgroundPrimary,
+    contentDescription: String = stringResource(R.string.vtmn_cd_is_loading)
```
[the remark this answers is not in the document]
Comment (FDELAHA24) [reply skeleton/src/main/java/com/decathlon/vitamin/compose/skeleton/SkeletonMofifierExt.kt:30]: I added this contentDescription to be able to customize the talkback feedback when the loading view is displayed just to let the user know something is loading, but yes it's kinda weird

**Comment by dktbitrise** [comment]
### You can download the apk here :

**Comment by dktbitrise** [comment]
### You can download the apk here :

**Comment by dktbitrise** [comment]
### You can download the apk here :

**Comment by dktbitrise** [comment]
### You can download the apk here :

**Comment by dktbitrise** [comment]
### You can download the apk here :

**Comment by GerardPaligot** [comment]
You have a lot of lint errors with your contribution:
```bash
> Task :skeletons:ktlintMainSourceSetCheck FAILED
/bitrise/src/skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/VerticalShimmer.kt:22:1 Needless blank line(s) (no-consecutive-blank-lines)
/bitrise/src/skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/VitaminSkeletonColors.kt:21:1 Unexpected blank line(s) before "}" (no-blank-line-before-rbrace)
/bitrise/src/skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/VitaminSkeletonColors.kt:22:1 Needless blank line(s) (no-consecutive-blank-lines)
/bitrise/src/skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/VitaminSkeletons.kt:8:1 Unused import (no-unused-imports)
/bitrise/src/skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/VitaminSkeletons.kt:10:1 Unused import (no-unused-imports)
/bitrise/src/skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/VitaminSkeletons.kt:11:1 Unused import (no-unused-imports)
/bitrise/src/skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/VitaminSkeletons.kt:14:1 Unused import (no-unused-imports)
/bitrise/src/skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/VitaminSkeletons.kt:37:1 Needless blank line(s) (no-consecutive-blank-lines)
/bitrise/src/skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/VitaminSkeletons.kt:50:1 Unexpected blank line(s) before "}" (no-blank-line-before-rbrace)
/bitrise/src/skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/VitaminSkeletons.kt:69:1 Needless blank line(s) (no-consecutive-blank-lines)
```

File: skeletons/README.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+This component draw a rectangle. You can add a modifier, but you can also customize the colors and the corner radius of the shape.
+
+```kotlin
+VitaminSkeleton.Rectangular()
+```
+
+Parameters | Descriptions
+-- | --
+`modifier: Modifier = Modifier` | The `Modifier` to be applied to the component
+`colors: SkeletonColors = VitaminSkeletonColors.primary()` | colors The colors of the background and the highlight effect
```
Comment (GerardPaligot) [inline skeletons/README.md:39]: remove `colors` keyword at the beginning

Comment (GerardPaligot) [reply skeletons/README.md:39]: same for all variants

File: skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/SkeletonColors.kt [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package com.decathlon.vitamin.compose.skeletons
+
+import androidx.compose.runtime.Immutable
+import androidx.compose.ui.graphics.Color
+
+@Immutable
+class SkeletonColors(
+    val background: Color,
```
Comment (GerardPaligot) [inline skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/SkeletonColors.kt:8]: In all modules, we have `backgroundColor`. Probably a good idea to rename it and maybe add this rule in the `CONVENTIONS.md` file in this PR or in a next contribution?

File: skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/VitaminSkeletonColors.kt [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+import androidx.compose.ui.graphics.Color
+import com.decathlon.vitamin.compose.foundation.VitaminTheme
+
+object VitaminSkeletonColors {
+    @Composable
+    fun primary(
+        background: Color = VitaminTheme.colors.vtmnBackgroundTertiary,
+        highlight: Color = VitaminTheme.colors.vtmnBackgroundPrimary
+    ): SkeletonColors = remember(
+        background
```
Comment (GerardPaligot) [inline skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/VitaminSkeletonColors.kt:14]: You need to add `highlight` too.

File: skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/VitaminSkeletonColors.kt [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.graphics.Color
+import com.decathlon.vitamin.compose.foundation.VitaminTheme
+
+object VitaminSkeletonColors {
+    @Composable
+    fun primary(
+        background: Color = VitaminTheme.colors.vtmnBackgroundTertiary,
+        highlight: Color = VitaminTheme.colors.vtmnBackgroundPrimary
```
Comment (GerardPaligot) [inline skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/VitaminSkeletonColors.kt:12]: Rename `background` and `highlight` to `backgroundColor` and `highlightColor`.

**Review by GerardPaligot — CHANGES_REQUESTED** [review]

File: skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/VitaminSkeletonColors.kt (hunk 7 above)
Comment (FDELAHA24) [reply skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/VitaminSkeletonColors.kt:14]: my bad

File: skeletons/README.md (hunk 5 above)
Comment (FDELAHA24) [reply skeletons/README.md:39]: alright

File: skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/SkeletonColors.kt (hunk 6 above)
Comment (FDELAHA24) [reply skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/SkeletonColors.kt:8]: good idea

File: skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/VitaminSkeletonColors.kt (hunk 8 above)
Comment (FDELAHA24) [reply skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/VitaminSkeletonColors.kt]: done

**Comment by dktbitrise** [comment]
### You can download the apk here :

**Comment by dktbitrise** [comment]
### You can download the apk here :
**sample-debug.apk :** 
https://app.bitrise.io/app/e9ae480151be4514/build/ccfc38d2-5f43-44b4-b8dc-65fca27abb96/artifact/e0a37ef4d78ef3f3/p/ed41324ba7fce2fb480f4cd2a8ae92e2

File: skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/SkeletonColors.kt (hunk 6 above)
Comment (GerardPaligot) [reply skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/SkeletonColors.kt:8]: You've updated the documentation but not the code here x)

File: skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/SkeletonColors.kt [hunk 9]
```diff
@@ -0,0 +1,10 @@
+package com.decathlon.vitamin.compose.skeletons
+
+import androidx.compose.runtime.Immutable
+import androidx.compose.ui.graphics.Color
+
+@Immutable
+class SkeletonColors(
```
Comment (GerardPaligot) [inline skeletons/src/main/java/com/decathlon/vitamin/compose/skeletons/SkeletonColors.kt:7]: Everywhere it is a dataclass to be comparable. You can use it here too?

**Review by GerardPaligot — CHANGES_REQUESTED** [review]

**Comment by dktbitrise** [comment]
### You can download the apk here :
**sample-debug.apk :** 
https://app.bitrise.io/app/e9ae480151be4514/build/04838b06-0069-40b4-9492-6004aa812092/artifact/d149eb9e2a604c29/p/be4018bda39d921ee17c248d1cb07ee3

**Review by GerardPaligot — APPROVED** [review]

**Pull request merged**